src/Castle.Core/DynamicProxy/ProxyGenerator.cs

```diff
@@ -595,7 +595,10 @@ public object CreateInterfaceProxyWithTarget(Type interfaceToProxy, object targe
 						var result = Marshal.QueryInterface(iUnknown, ref interfaceId, out interfacePointer); // Increment the reference count
 						var isInterfacePointerNull = interfacePointer == IntPtr.Zero;
 						Marshal.Release(iUnknown); // Decrement the reference count
-						Marshal.Release(interfacePointer); // Decrement the reference count
+						if (!isInterfacePointerNull)
+						{
+							Marshal.Release(interfacePointer); // Decrement the reference count
+						}
 
 						if (result == 0 && isInterfacePointerNull)
```

Comment:
Looks like all the error cases weren't handled in this code. This code path probably hasn't been used much.